src/exchange/apiService.ts

```diff
@@ -135,12 +144,16 @@ export class APIService {
   private transformRequest = (data: any, headers?: any) => {
     if (this.apiToken) {
       headers["x-api-token"] = this.apiToken;
-
     }
     else {
       headers.Authorization = `Bearer ${this.token}`;
     }
 
+    if(this.uuid && this.uuid!="")
+    {
+        headers["x-mm-id"] = this.uuid;
+    }
+    
```

Comment:
Previously I was setting default UUID as "uuid-default" in configs. But while making these changes I am thinking to not send x-mm-id UUID at all if it's not defined or set to "". I believe an empty uuid might also be considered as a unique UUID and would be assigned a dedicated rate limit. Let me still confirm with Memo if that's not the case then we can safely have an "" UUID as default and keep sending "" in headers as well.

Reply:
Any update on the check here @NidaKarimAli ?

Reply:
I have resolved the comments on both the PRs. The common one which you had was regarding not checking uuid="" while setting it to x-mm-id in headers. This is something just recently caught my attention that even keeping a "" x-mm-id header would be treated as unique UUID in WAF and will assign users with custom rate limits despite them not setting their provided uuid at all. So talked to @MemoAlfa about it and for now, if uuid is "", I wont be setting `x-mm-id` in the header. In future this would become a mandatory requirement for colocation partners so requests with no x-mm-id header will be rejected.